mist-api/src/lib/util.js

```diff
@@ -0,0 +1,59 @@
+'use strict'
+
+// Vendor
+const _ = require('lodash')
+
+// Local
+const util = require('core-lib/util')
+
+exports.splitIntoQueryTerms = (value) => {
+    return util.splitIntoTerms(value).map((term) => `%${term}%`)
+}
+
+exports.checkQueryAndTarget = (queryValue, target) => {
```

Comment:
This function really does two things that really should not be coupled together (except for the fact that it appears DRY). If `target` is undefined, we have a problem with our code - not the user request. Please convert these to `assert` checks within their respective function.

Let's scrap this function altogether and do the following refactor:

```
exports.assignExactMatchCriteria = (queryValue, target, fields) => {
  queryValue = queryValue && queryValue.trim()
  assert(target)
  ...
}

exports.assignInexactMatchCriteria = (queryValue, target, fields) => {
  queryValue = queryValue && queryValue.trim()
  assert(target)
  ...
}
```

It looks unDRY, but each function should really be responsible for checking its own arguments and it doesn't feel right to have a separate function to simply trim the queryValue.

Going forward, probably will want to move this logic into a middleware at some point.

Reply:
Done. I didn't like it too.